gcs/constraints/circuit/circuit_scc.cc

```diff
@@ -0,0 +1,1250 @@
+                    p_line << "p ";
+                    p_line << shifted_pos_geq[node][count].backwards_reif_line << " ";
+                    p_line << pos_var_data.at(node).plus_one_lines.at(next_node).leq_line << " + s";
+                    proof.emit_proof_line(p_line.str(), ProofLevel::Temporary);
+
+                    emit_proof_comment_if_enabled("Next implies: succ[" + to_string(node) + "] = " + to_string(next_node) + " and " +
+                            shifted_pos_eq[node][count - 1].comment_name + " = " + to_string(count - 1) + " => 0 >= 1",
+                        options, state);
+
+                    successor_implies_line = proof.emit_rup_proof_line(WeightedPseudoBooleanSum{} + 1_i * (! shifted_pos_eq[node][count - 1].flag) + 1_i * (succ[node] != Integer{next_node}) >= 1_i, ProofLevel::Current);
+                    // proof.emit_proof_line("fail", ProofLevel::Top);
+                }
+            }});
+        }
+        else {
+            // Not using shifted pos, just use the actual pos values
+            state.infer_true(JustifyExplicitly{[&](Proof & proof) {
+                // succ[node] == next_node /\ pos[node] == count - 1 => pos[next_node] == count
+                successor_implies_line = proof.emit_rup_proof_line(
+                    WeightedPseudoBooleanSum{} + 1_i * (pos_var_data.at(node).var != Integer{count - 1}) +
+                            1_i * (succ[node] != Integer{next_node}) + 1_i * (pos_var_data.at(next_node).var == Integer{count}) >=
+                        1_i,
+                    ProofLevel::Current);
+            }});
+        }
+
+        return successor_implies_line;
+    }
+
+    auto prove_not_same_val(const long & root, const long & middle, const long & next_node, const long & count,
+        map<long, ShiftedPosDataMaps> & flag_data, const PosVarDataMap & pos_var_data, PosAllDiffData & pos_alldiff_data,
+        const vector<IntegerVariableID> & succ, const SCCOptions & options, State & state)
+    {
+        auto succesor_implies_not_mid_line = ProofLine{};
+        // successor_implies_line :=
+        // Prove (shift)pos[next_node][count] => ! (shift)pos[mid][count]
+        create_shifted_pos(root, middle, count, flag_data[root], pos_var_data, pos_alldiff_data,
+            succ, options, state);
+        emit_proof_comment_if_enabled("Successor implies not mid", options, state);
+        auto n = static_cast<long>(succ.size());
+        if (root != 0) {
+            create_flag_for_greater_than(next_node, middle, flag_data[next_node], pos_var_data, pos_alldiff_data,
+                succ, options, state);
+
+            PLine temp_p_line;
+
+            auto & shifted_pos_geq = flag_data[root].shifted_pos_geq;
+            auto & shifted_pos_eq = flag_data[root].shifted_pos_eq;
+
+            state.infer_true(JustifyExplicitly{[&](Proof & proof) {
+                // TODO: All these should be ProofLevel::Temporary, but unfortunately that fails in specific instances
```

Comment:
Did you ever figure that out?

Reply:
Yes, the way I was emitting comments was messing with the proof levels. Should be fixed now.